test/integration/imu_system.cc

```diff
@@ -250,3 +295,285 @@ TEST_F(ImuTest, IGN_UTILS_TEST_DISABLED_ON_WIN32(OrientationDisabled))
+  auto topicDEFAULT = "/imu_test_DEFAULT";
+
+  // subscribe to imu topic
+  transport::Node node;
+  node.Subscribe(topicENU, &imuENUCb);
+  node.Subscribe(topicNED, &imuNEDCb);
+  node.Subscribe(topicNWU, &imuNWUCb);
+  node.Subscribe(topicCUSTOM, &imuCUSTOMCb);
+  node.Subscribe(topicDEFAULT, &imuDEFULTCb);
+
+  // Run server
+  server.Run(true, 200u, false);
+
+  // Check we received messages
+  EXPECT_TRUE(lastImuMsgENU.has_orientation());
+  EXPECT_TRUE(lastImuMsgNED.has_orientation());
+  EXPECT_TRUE(lastImuMsgNWU.has_orientation());
+  EXPECT_TRUE(lastImuMsgCUSTOM.has_orientation());
+  EXPECT_TRUE(lastImuMsgDEFAULT.has_orientation());
+
+  // For the DEFAULT msg, orientation is reported relative
+  // to the original pose, which should be identity quaternion.
+  EXPECT_NEAR(lastImuMsgDEFAULT.orientation().x(), 0, 1e-2);
+  EXPECT_NEAR(lastImuMsgDEFAULT.orientation().y(), 0, 1e-2);
+  EXPECT_NEAR(lastImuMsgDEFAULT.orientation().z(), 0, 1e-2);
+  EXPECT_NEAR(lastImuMsgDEFAULT.orientation().w(), 1, 1e-2);
+
+  // For the ENU msg
+  EXPECT_NEAR(lastImuMsgENU.orientation().x(), 0, 1e-2);
+  EXPECT_NEAR(lastImuMsgENU.orientation().y(), 0, 1e-2);
+  EXPECT_NEAR(lastImuMsgENU.orientation().z(), 0.707, 1e-2);
+  EXPECT_NEAR(lastImuMsgENU.orientation().w(), 0.707, 1e-2);
+
+  // For the NED msg
+  EXPECT_NEAR(lastImuMsgNED.orientation().x(), -1, 1e-2);
+  EXPECT_NEAR(lastImuMsgNED.orientation().y(), 0, 1e-2);
+  EXPECT_NEAR(lastImuMsgNED.orientation().z(), 0, 1e-2);
+  EXPECT_NEAR(lastImuMsgNED.orientation().w(), 0, 1e-2);
+
+  // For the NWU msg
+  EXPECT_NEAR(lastImuMsgNWU.orientation().x(), 0, 1e-2);
+  EXPECT_NEAR(lastImuMsgNWU.orientation().y(), 0, 1e-2);
+  EXPECT_NEAR(lastImuMsgNWU.orientation().z(), 0, 1e-2);
+  EXPECT_NEAR(lastImuMsgNWU.orientation().w(), 1, 1e-2);
+
+  // For the CUSTOM msg
+  EXPECT_NEAR(lastImuMsgCUSTOM.orientation().x(), -0.5, 1e-2);
+  EXPECT_NEAR(lastImuMsgCUSTOM.orientation().y(), 0.5, 1e-2);
+  EXPECT_NEAR(lastImuMsgCUSTOM.orientation().z(), 0.5, 1e-2);
+  EXPECT_NEAR(lastImuMsgCUSTOM.orientation().w(), 0.5, 1e-2);
+}
+
+/////////////////////////////////////////////////
+// The test checks if orientations are reported correctly for a rotating body.
+// The world includes a sphere rolling down a plane, with axis of rotation
+// as the "west" direction vector, using the right hand rule.
+TEST_F(ImuTest, IGN_UTILS_TEST_DISABLED_ON_WIN32(RotatingBody))
+{
+  imuMsgs.clear();
+  clearLastImuMsgs();
+
+  // Start server
+  ServerConfig serverConfig;
+  const auto sdfFile = common::joinPaths(std::string(PROJECT_SOURCE_PATH),
+    "test", "worlds", "imu_rotating_demo.sdf");
+  serverConfig.SetSdfFile(sdfFile);
+
+  Server server(serverConfig);
+  EXPECT_FALSE(server.Running());
+  EXPECT_FALSE(*server.Running(0));
+
+  msgs::IMU currentImuMsgDefault_1;
+  msgs::IMU currentImuMsgDefault_2;
+  msgs::IMU currentImuMsgDefault_3;
+
+  auto defaultCb1 = std::function<void(const msgs::IMU &_msg)>(
```

Comment:
@adityapande-1995 nit^2: it might ready better if you do:
```suggestion
  std::function<void(const msgs::IMU &_msg)> defaultCb1 = 
```
instead. Same below.

Reply:
Done